compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/intrinsics/IteratorNext.kt

```diff
@@ -31,7 +32,7 @@ import org.jetbrains.org.objectweb.asm.Type
 object IteratorNext : IntrinsicMethod() {
 
     override fun toCallable(expression: IrFunctionAccessExpression, signature: JvmMethodSignature, context: JvmBackendContext): IrIntrinsicFunction {
-        val type = AsmUtil.unboxType(signature.returnType)
+        val type = if (expression.type.isPrimitiveType()) signature.returnType else AsmUtil.unboxType(signature.returnType)
```

Comment:
@juan-chen The aim of this change is not clear, could you explain a bit

Reply:
Added a comment.

Reply:
@juan-chen Could the fix be made in `AsmUtil.unboxType()`? i.e., make it a no-op if it's a primitive. Or would that cause breaks somewhere else?

Reply:
AsmUtil.unboxType is a low-level function. It might be riskier to change.